### PR TITLE
[documentation]  Add UML diaram for Backhaul STA steering

### PR DESCRIPTION
**Pull request by VladyslavTupikin**
According to the test plan and source code
create the UML diagram which describes
data flow and methods for Backhaul STA steering.

### CODE REVIEW

File: documentation/images/plantuml/agent/flows/sta_steering.puml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@startuml
+->controller: dev_reset_default
+->controller: Prepare setup( execute 4.1.2 test)
+controller->backhaul_manager: send_cmdu(Backhaul Steering Request)
+backhaul_manager->backhaul_manager: handle_1905_msg(\n Backhaul Steering Request message,\n Backhaul Steering Request TLV(0x9E))
+backhaul_manager->controller: send_cmdu(ACK)
+backhaul_manager->controller: send_cmdu(Associated Request)
+controller->controller: \nhandle_1905_msg(Associated Request)
+controller->backhaul_manager: send_cmdu(Associated Response)
+backhaul_manager->backhaul_manager: handle_1905_msg(Associated Response)
```
Comment (arnout) [inline documentation/images/plantuml/agent/flows/sta_steering.puml]: As mentioned orally:
- it's not a 1905 message, so not `send_cmdu` and `handle_1905_msg`; rather, it should go through bwl;
- it's not to the controller, but to the bAP to which it's connected;
- it's Association, not Associated.

**Review by arnout — CHANGES_REQUESTED** [review]

File: documentation/images/plantuml/agent/flows/sta_steering.puml (hunk 1 above)
Comment (VladyslavTupikin) [reply documentation/images/plantuml/agent/flows/sta_steering.puml]: @arnout fixed. Updated the branch.

File: documentation/images/plantuml/agent/flows/bh_sta_steering.puml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+backhaul_manager->backhaul_manager: handle_1905_msg(\n Backhaul Steering Request message,\n Backhaul Steering Request TLV(0x9E))
+backhaul_manager->controller: send_cmdu(ACK)
+backhaul_manager->backhaul_manager: sta_wlan_hal->connect()
+backhaul_manager->wpa_supplicant: NETWORK_ENABLE
+wpa_supplicant->ctt_agent_2: Association Request
+ctt_agent_2->wpa_supplicant: Association Response
+wpa_supplicant->backhaul_manager: ENABLED
+backhaul_manager->controller: send_cmdu(Backhaul Steering Response message)
+controller->controller: handle_1905_msg(Backhaul Steering Response message)
+controller->backhaul_manager: send_cmdu(Backhaul Steering Request)
```
Comment (arnout) [inline documentation/images/plantuml/agent/flows/bh_sta_steering.puml]: Maybe add a note that this is with invalid channel.

File: documentation/images/plantuml/agent/flows/bh_sta_steering.puml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+@startuml
+->controller: dev_reset_default
+->controller: Prepare setup( execute 4.1.2 test)
+controller->backhaul_manager: send_cmdu(Backhaul Steering Request)
+backhaul_manager->backhaul_manager: handle_1905_msg(\n Backhaul Steering Request message,\n Backhaul Steering Request TLV(0x9E))
+backhaul_manager->controller: send_cmdu(ACK)
+backhaul_manager->backhaul_manager: sta_wlan_hal->connect()
```
Comment (arnout) [inline documentation/images/plantuml/agent/flows/bh_sta_steering.puml]: Could be useful to clarify:
```suggestion
backhaul_manager->backhaul_manager: sta_wlan_hal->connect(bssid, channel)
```

**Review by arnout — APPROVED** [review]

**Comment by arnout** [comment]
We should probably have a README.md with the design details as well, though there's really not much to say I guess.

**Review by tomereli — CHANGES_REQUESTED** [review]
Please review https://github.com/prplfoundation/prplMesh/pull/1423 first and update the diagram accordingly. Actually, this diagram should be added to #1423 and this PR should be closed.

**Review by tomereli — APPROVED** [review]
Approving, will be updated in #1423

File: documentation/images/plantuml/agent/flows/bh_sta_steering.puml (hunk 2 above)
Comment (VladyslavTupikin) [reply documentation/images/plantuml/agent/flows/bh_sta_steering.puml]: fixed

File: documentation/images/plantuml/agent/flows/bh_sta_steering.puml (hunk 3 above)
Comment (VladyslavTupikin) [reply documentation/images/plantuml/agent/flows/bh_sta_steering.puml]: fixed

**Pull request merged**